internal/kojiapi/server.go

```diff
@@ -223,34 +229,24 @@ func splitExtension(filename string) string {
 	return "." + strings.Join(filenameParts[1:], ".")
 }
 
-func composeStateToStatus(state common.ComposeState) string {
-	switch state {
-	case common.CFailed:
-		return "failure"
-	case common.CFinished:
-		return "success"
-	case common.CRunning:
-		return "pending"
-	case common.CWaiting:
-		return "pending"
-	default:
-		panic("invalid compose state")
+func composeStatusFromJobStatus(js *worker.JobStatus) string {
+	if js.Canceled {
+		return StatusFailure
+	}
+
+	if js.Started.IsZero() {
+		return StatusPending
+	}
+
+	if js.Finished.IsZero() {
+		return StatusRunning
```

Comment:
This is a change of behavior. Is that intended? Did the koji openapi spec not need adjusting like the cloud one?

Reply:
Indeed! Thanks for the catch.